### PR TITLE
Only expose SHA1 streaming api for Mozilla dist

### PR DESCRIPTION
**Pull request by R1kM**
Follow-up of #638. The previous PR was extracting all Streaming hash APIs, including some for hashes not included in dist/mozilla.
This led to some extracted files not being caught in Krml bundles, and to spurious files with extremely long names that led to failures on Windows (see the windows CI for #640).
This PR attempts to fix that by restricting the extraction of the streaming hash API for Mozilla to SHA1.
It also includes locally regenerated dist/mozilla files.

### CODE REVIEW

**Comment by Frosne** [comment]
I have already seen the problem you are talking about. It seems that the main offender is Blake. 

What if we do not extract the Blake family? And leave everything else?

**Comment by R1kM** [comment]
Sounds good. Should we include everything related to hashes then?

**Comment by Frosne** [comment]
@protz, might adding [rename=Hacl_Blake] help with the names?

**Comment by Frosne** [comment]
> Sounds good. Should we include everything related to hashes then?

yes, I think so.

**Comment by R1kM** [comment]
Ok. Do you mind if we do that in a subsequent PR so that we can merge this one quickly and unblock CI for the dist refresh?

**Comment by Frosne** [comment]
Sure

**Review by Frosne — APPROVED** [review]

**Comment by msprotz** [comment]
@Frosne it might be worth exploring if you can just cherry-pick individual files from the dist/gcc-compatible folder rather than modifying the bundles every time. Concretely, why do you need dist/mozilla? If I look at, say, Hacl_Poly1305_128.c for instance, the files are identical in both directories.

This might make the whole integration process easier for you.

**Review by msprotz — APPROVED** [review]
This is fine to unbreak the build, but I think we need to urgently switch to a model where

i) PRs from Mozilla are submitted *properly*, with a suitable dist refresh. Is this something we can automate?
ii) Mozilla directly consumes files from the dist/gcc-compatible folder rather than submitting PRs one by one, with a high danger of build breakage

Let's unbreak the build with this one, and then let's assess what we can do in the long run. Thanks.

**Comment by Frosne** [comment]
Could we just extract all the code to dist/mozilla? I totally see your
point about the annoying PRs all the time :)

**Comment by msprotz** [comment]
That would be great. But what would be the difference between dist/mozilla and dist/gcc-compatible then? Are there Mozilla-specific options? I don't think so...

**Comment by msprotz** [comment]
A long while ago, there was a problem with excessive inclusion of headers, which meant that if you only used a subset of the files, you might end up packaging more than you needed. But that was fixed a long while ago and it was, I think, the only reason why there was a separate mozilla folder

**Comment by karthikbhargavan** [comment]
Mozilla is an important consumer of HACL* and I wouldn't mind us maintaining dist/mozilla, but it is a good idea to recall the precise requirements for this distribution. Having a single directory like dist/gcc-compatible would be much simpler to maintain, unless there were good reason why this doesn't work?

File: dist/mozilla/Hacl_Hash_SHA1.c [hunk 1]
```diff
@@ -142,7 +154,7 @@ static void legacy_pad(uint64_t len, uint8_t *dst)
   store64_be(dst3, len << (uint32_t)3U);
 }
 
-static void legacy_finish(uint32_t *s, uint8_t *dst)
+void Hacl_Hash_Core_SHA1_legacy_finish(uint32_t *s, uint8_t *dst)
```
[the remark this answers is not in the document]
Comment (R1kM) [reply dist/mozilla/Hacl_Hash_SHA1.c:157]: Yeah, that's already what is done, but somehow this doesn't seem sufficient. Let's debug it in a further PR

**Comment by R1kM** [comment]
Merging to unblock the build and dist refresh. Let's continue discussing what a better workflow would be here.

**Pull request merged**